src/libsystemd/sd-device/device-trigger.c

```diff
@@ -0,0 +1,483 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdbool.h>
+
+#include "sd-device.h"
+
+#include "alloc-util.h"
+#include "device-enumerator-private.h"
+#include "device-private.h"
+#include "device-trigger-private.h"
+#include "device-util.h"
+#include "fileio.h"
+#include "id128-util.h"
+#include "hashmap.h"
+#include "set.h"
+#include "string-util.h"
+#include "time-util.h"
+
+#define SYNTH_UUID_KEY       "SYNTH_UUID"
+#define SOURCE_KEY           "SOURCE"
+#define DEFAULT_SOURCE_VALUE "SD"
+
+struct sd_device_trigger {
+        unsigned n_ref;
+
+        sd_device **devices;
+        size_t n_devices, n_allocated;
+
+        const char *action;
+        char uuid[SD_ID128_UUID_STRING_MAX];
```

Comment:
please store parsed data only. we don't want ambiguities around uppercase/lowecase and such. Please use sd_id128_t here, and format when needed only.